src/valid/interface.rs

```diff
@@ -332,7 +332,10 @@ impl super::Validator {
         use super::TypeFlags;
 
         log::debug!("var {:?}", var);
-        let type_info = &self.types[var.ty.index()];
+        let type_info = self
+            .types
+            .get(var.ty.index())
+            .ok_or(GlobalVariableError::InvalidType)?;
```

Comment:
Here, too, the error message for `InvalidType` is `"Type isn't compatible with the storage class"`, which is just bad.